tests/lava/proc/io/test_dataloader.py

```diff
@@ -0,0 +1,229 @@
+@requires(CPU)
+@tag('fixed_pt')
+class PyDummyProc(PyLoihiProcessModel):
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, int)
+    state: np.ndarray = LavaPyType(np.ndarray, int)
+
+    def run_spk(self) -> None:
+        self.s_out.send(self.state)
+
+
+class TestDataloader(unittest.TestCase):
+    def test_state_loader(self) -> None:
+        """Tests state dataloder"""
+        num_steps = 30
+        shape = (5, 7)
+        interval = 5
+        offset = 2
+
+        proc = DummyProc(shape)
+        dataloader = State(DummyDataset(shape), interval, offset)
+        gt = ReceiveProcess(shape=dataloader.gt.shape, buffer=num_steps)
+        out = ReceiveProcess(shape=shape, buffer=num_steps)
+        dataloader.connect_var(proc.state)
+        dataloader.gt.connect(gt.a_in)
+        proc.s_out.connect(out.a_in)
+
+        run_condition = RunSteps(num_steps=num_steps)
+        run_config = TestRunConfig(select_tag='fixed_pt')
+        proc.run(condition=run_condition, run_cfg=run_config)
+        gt_data = gt.data.get()
+        out_data = out.data.get()
+        proc.stop()
+
+        dataset = DummyDataset(shape)
+        for i in range(offset + 1, num_steps):
+            id = (i - offset - 1) // interval
+            data, gt = dataset[id]
+            self.assertTrue(np.array_equal(gt, gt_data[..., i].item()))
+            self.assertTrue(
+                np.array_equal(data, out_data[..., i]),
+                f'Expected data and out_data at {i=} to be same. '
+                f'Found {data=} and {out_data[..., i]=}'
+            )
+
+    def test_spike_loader(self) -> None:
+        """Tests spike dataloder"""
+        shape = (5, 7)
+        interval = 5
+        offset = 2
+        num_steps = 31 + offset
+
+        dataloader = Spike(SpikeDataset(shape + (interval,)), interval, offset)
+        gt = ReceiveProcess(shape=dataloader.gt.shape, buffer=num_steps)
+        out = ReceiveProcess(shape=shape, buffer=num_steps)
+        dataloader.gt.connect(gt.a_in)
+        dataloader.s_out.connect(out.a_in)
+
+        run_condition = RunSteps(num_steps=num_steps)
+        run_config = TestRunConfig(select_tag='fixed_pt')
+        dataloader.run(condition=run_condition, run_cfg=run_config)
+        gt_data = gt.data.get()
+        out_data = out.data.get()
+        dataloader.stop()
+
+        dataset = SpikeDataset(shape + (interval,))
+        for i in range(offset + 1, num_steps, interval):
+            id = (i - offset - 1) // interval
+            data, gt = dataset[id]
+            gt_error = np.abs(gt_data[..., i:i + interval] - gt).sum()
+            spike_error = np.abs(out_data[..., i:i + interval] - data).sum()
+            self.assertTrue(gt_error == 0)
+            self.assertTrue(
+                spike_error == 0,
+                f'Expected data and out_data at {i=} to be same. '
+                f'Found {data=} and {out_data[..., i:i + interval]=}'
+            )
+
+    def test_spike_loader_less_steps(self) -> None:
```

Comment:
There is quite a lot of code duplication in these tests. Can you put the setup and assertions inside a function and call this from different tests with different parameters?

Reply:
Thanks for pointing it out. Refactored it.